### PR TITLE
[CHERRY-PICK]fix(p2p): delete all preheat policies before delete project

### PR DESCRIPTION
**Pull request by chlins**
Closes:  https://github.com/goharbor/harbor/issues/12891 

Signed-off-by: chlins <chlins.zhang@gmail.com>

### CODE REVIEW

File: src/server/v2.0/handler/project.go [hunk 1]
```diff
@@ -212,6 +215,11 @@ func (a *projectAPI) DeleteProject(ctx context.Context, params operation.DeleteP
 		}
 	}
 
+	// preheat policies under the project should be deleted before deleting the project
```
Comment (heww) [inline src/server/v2.0/handler/project.go:218]: `before` -> `after`

File: src/controller/p2p/preheat/controller.go [hunk 2]
```diff
@@ -419,6 +421,21 @@ func (c *controller) DeletePolicy(ctx context.Context, id int64) error {
...
+		return err
+	}
+
+	for _, p := range policies {
+		if err = c.DeletePolicy(ctx, p.ID); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```
Comment (heww) [inline src/controller/p2p/preheat/controller.go:437]: How many SQLs will be executed here? Can we do this by one SQL?

File: src/server/v2.0/handler/project.go (hunk 1 above)
Comment (chlins) [reply src/server/v2.0/handler/project.go]: changed

File: src/controller/p2p/preheat/controller.go (hunk 2 above)
Comment (chlins) [reply src/controller/p2p/preheat/controller.go:437]: @heww After review the code, i think we will not have too many policies under the project, and in DeletePolicy, we do extra clean work like unschedule scheduler and delete executions for one policy. So keep the logic.

**Review by heww — APPROVED** [review]

**Review by steven-zou — APPROVED** [review]
lgtm

**Pull request merged**